apps/fft/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+cmake_minimum_required(VERSION 3.16)
+project(fft)
+
+enable_testing()
+
+# Set up language settings
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED YES)
+set(CMAKE_CXX_EXTENSIONS NO)
+
+# Find Halide
+find_package(Halide REQUIRED)
+
+# Generator
+add_executable(fft.generator fft_generator.cpp fft.cpp)
+target_link_libraries(fft.generator PRIVATE Halide::Generator)
+
+# Filters
+add_halide_library(fft_forward_r2c FROM fft.generator
+                   GENERATOR fft
+                   PARAMS direction=samples_to_frequency size0=16 size1=16 gain=0.00390625 input_number_type=real output_number_type=complex)
```

Comment:
style nit: might be nice to split the PARAMS into multiple lines rather than one loooong one, if easy to do